File: SistemaWeb/app/Models/Notificacion.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Notificacion extends Model
{
    use HasFactory;
    protected $table = 'notificaciones';

    protected $fillable = [
        'id',
        'titulo',
        'descripcion',
        'fecha',
        'grupo_id',
        'created_at'
    ];

    public function grupo() {
        return $this->belongsTo(Grupo::class, 'grupo_id');
    }
}

File: AppMovil/lib/models/notificacion.dart
class Notificacion {
  num _id;
  String _titulo;
  String _descripcion;
  dynamic _enviado;

  get id => this._id;
  get titulo => this._titulo;
  get descripcion => this._descripcion;
  get enviado => this._enviado;

  Notificacion(this._id, this._titulo, this._descripcion, this._enviado);

  factory Notificacion.fromJson(Map<dynamic, dynamic> data) {
    return Notificacion(
      data['id'],
      data['titulo'],
      data['descripcion'],
      data['created_at'],
    );
  }
}
